src/prism.c

```diff
@@ -12647,25 +12663,32 @@ parse_variable_call(pm_parser_t *parser) {
         }
 
         if (!parser->current_scope->closed && pm_token_is_numbered_parameter(parser->previous.start, parser->previous.end)) {
-            // Indicate that this scope is using numbered params so that child
-            // scopes cannot.
-            parser->current_scope->numbered_params = true;
```

Comment:
This optimization was here so that `outer_scope_using_numbered_params_p` didn't have to scan too many scopes, it would already be cached. Can we bring this back?

Reply:
I'm not sure I'm following - what part of this did we lose? My understanding is we're still setting `parser->current_scope->numbered_parameters` to a non-zero value, so all the same cases will still return true from `outer_scope_using_numbered_parameters_p`